index.js

```diff
@@ -70,283 +70,314 @@ Fs.readdirSync(Path.join(__dirname, 'helpers')).forEach(function (file) {
 * @param {assemblerGetTemplates} assembler.getTemplates - Method to assemble templates
 * @param {assemblerGetTranslations} assembler.getTranslations - Method to assemble translations
 */
-function Paper(settings, themeSettings, assembler) {
-    var self = this;
+class Paper {
+    constructor(settings, themeSettings, assembler) {
+    let self = this;
 
-    self.handlebars = Handlebars.create();
+        self.handlebars = Handlebars.create();
 
-    self.handlebars.templates = {};
-    self.translator = null;
-    self.inject = {};
-    self.decorators = [];
+        self.handlebars.templates = {};
+        self.translator = null;
+        self.inject = {};
+        self.decorators = [];
 
-    self.settings = settings || {};
-    self.themeSettings = themeSettings || {};
-    self.assembler = assembler || {};
+        self.settings = settings || {};
+        self.themeSettings = themeSettings || {};
+        self.assembler = assembler || {};
 
-    _.each(helpers, function (helper) {
-        helper(self);
-    });
-}
+        _.each(helpers, function (helper) {
+            helper(self);
+        });
+    }
 
-/**
- * Renders a string with the given context
- * @param  {String} string
- * @param  {Object} context
- */
-Paper.prototype.renderString = function (string, context) {
-    return this.handlebars.compile(string)(context);
-};
+    /**
+    * Paper factory constructor. This constructor preloads the translation files and appends the main 
+    * context to the paper instance for use in the helpers.
+    *
+    * @param {Object} data - Theme settings (configuration)
+    * @param {Object} assembler - Assembler with getTemplates and getTranslations methods.
+    *
+    */
+    static createInstance(data, assembler) {
+        let translator;
+
+        return new Promise((resolve, reject) => {
```

Comment:
Can you use a callback instead? hapi has some issues with promises. https://github.com/hapijs/hapi/issues/2097